### ISSUE TITLE
[enhancement] Migrate psalm to use symfony console command?

### ISSUE THREAD

**Issue by mr-feek** [vimeo/psalm#3383]
`symfony/console` is already a dependency of this project that was introduced when plugins were. Has there been any thought in migrating psalm to be a symfony command? Would a PR doing so  be welcome?

I would think it might make maintenance a little easier, but I have no idea how intricate the migration would be, and how hard it would be to adjust the various output formats.

**Comment by weirdan** [comment]
One benefit of this I can think of is that it would be possible to automatically generate completion scripts for shells: https://github.com/stecman/symfony-console-completion

On the negative side, Input (options) and output formats may need to change (BC break).

**Comment by mr-feek** [comment]
+1 for https://github.com/stecman/symfony-console-completion, that's actually what prompted me to look for this enhancement!

**Comment by muglug** [comment]
yeah, definite BC break, so this would be for v4. As long as running just `vendor/bin/psalm` would still run analysis, I'm happy.
